homeassistant/components/binary_sensor/rfxtrx.py

```diff
@@ -0,0 +1,234 @@
+
+        device = RfxtrxBinarySensor(event, entity[ATTR_NAME],
+                                    entity[CONF_SENSOR_CLASS],
+                                    entity[ATTR_FIREEVENT],
+                                    entity[ATTR_OFF_DELAY],
+                                    entity[ATTR_DATA_BITS],
+                                    entity[CONF_COMMAND_ON],
+                                    entity[CONF_COMMAND_OFF])
+        device.hass = hass
+        sensors.append(device)
+        rfxtrx.RFX_DEVICES[device_id] = device
+
+    add_devices_callback(sensors)
+
+    # pylint: disable=too-many-branches
+    def binary_sensor_update(event):
+        """Callback for control updates from the RFXtrx gateway."""
+        if not isinstance(event, rfxtrxmod.ControlEvent):
+            return
+
+        device_id = slugify(event.device.id_string.lower())
+
+        if device_id in rfxtrx.RFX_DEVICES:
+            sensor = rfxtrx.RFX_DEVICES[device_id]
+        else:
+            sensor = rfxtrx.get_pt2262_device(device_id)
+
+        if sensor is None:
+            # Add the entity if not exists and automatic_add is True
+            if not config[ATTR_AUTOMATIC_ADD]:
+                return
+
+            poss_dev = rfxtrx.find_possible_pt2262_device(device_id)
+
+            if poss_dev is not None:
+                poss_id = slugify(poss_dev.event.device.id_string.lower())
+                _LOGGER.info("Found possible matching deviceid %s.",
+                             poss_id)
+
+            pkt_id = "".join("{0:02x}".format(x) for x in event.data)
+            sensor = RfxtrxBinarySensor(event, pkt_id)
+            rfxtrx.RFX_DEVICES[device_id] = sensor
+            add_devices_callback([sensor])
+            _LOGGER.info("Added binary sensor %s "
+                         "(Device_id: %s Class: %s Sub: %s)",
+                         pkt_id,
+                         slugify(event.device.id_string.lower()),
+                         event.device.__class__.__name__,
+                         event.device.subtype)
+
+        elif not isinstance(sensor, RfxtrxBinarySensor):
+            return
+        else:
+            _LOGGER.info("Binary sensor update "
+                         "(Device_id: %s Class: %s Sub: %s)",
+                         slugify(event.device.id_string.lower()),
+                         event.device.__class__.__name__,
+                         event.device.subtype)
+
+        if sensor.is_pt2262:
+            cmd = rfxtrx.get_pt2262_cmd(device_id, sensor.data_bits)
+            _LOGGER.info("applying cmd %s to device_id: %s)",
+                         cmd, sensor.masked_id)
+            sensor.apply_cmd(int(cmd, 16))
+        else:
+            if not sensor.is_on or sensor.should_fire_event:
+                sensor.update_state(True)
+
+        if (sensor.is_on and sensor.off_delay is not None and
+                sensor.delay_listener is None):
```

Comment:
I don't see how a non pt2262 binary sensor can receive the Command Off event (without using a delay)? Devices like the KaKu AMST-606 or the KaKu remote do send both On and Off Commands.